lib/rules/prefer-destructuring.js

```diff
@@ -56,6 +96,12 @@ module.exports = {
             if (typeof enabledTypes.object !== "undefined") {
                 checkObjects = enabledTypes.object;
             }
+
+            // checkArrays and checkObjects is set to false and use method shouldCheck instead
+            if (typeof enabledTypes.VariableDeclarator !== "undefined" || typeof enabledTypes.AssignmentExpression !== "undefined") {
+                checkArrays = false;
+                checkObjects = false;
+            }
```

Comment:
It's not a blocker for this PR, but having two separate methods of checking the options depending on the type of configuration used seems like a bug hazard if something changes later on. Maybe the old config-checking variables could be eliminated too with something like

```js
const normalizedOptions = typeof enabledTypes.array !== "undefined" || typeof context.options[0].object !== "undefined"
    ? { VariableDeclarator: enabledTypes, AssignmentExpression: enabledTypes }
    : enabledTypes
```

In other words, the old `array` and `object` schema would basically be transformed into a special case of the `VariableDeclarator`/`AssignmentExpression` schema where both assignment types happen to have the same configured options.